tikv-client/src/main/java/com/pingcap/tikv/expression/PartitionPruner.java

```diff
@@ -80,15 +80,27 @@ static void generateRangeExprs(
       if (current.equals("MAXVALUE")) {
         leftHand = "true";
       } else {
-        leftHand = String.format("%s < %s", partExprStr, current);
+        leftHand = String.format("%s < %s", wrapColumnName(partExprStr), current);
       }
       if (i == 0) {
         partExprs.add(parser.parseExpression(leftHand));
       } else {
         String previous = partInfo.getDefs().get(i - 1).getLessThan().get(lessThanIdx);
-        String and = String.format("%s and %s", partExprStr + ">=" + previous, leftHand);
+        String and =
+            String.format("%s >= %s and %s", wrapColumnName(partExprStr), previous, leftHand);
         partExprs.add(parser.parseExpression(and));
       }
     }
   }
+
+  private static String wrapColumnName(String columnName) {
+    if (columnName.startsWith("`") && columnName.endsWith("`")) {
+      return columnName;
+    } else if (columnName.contains("(") && columnName.contains(")")) {
```

Comment:
what if columnName = `a + b`?

Reply:
https://docs.pingcap.com/tidb/stable/partitioned-table#partitioning-limitations-relating-to-functions

![image](https://user-images.githubusercontent.com/5574887/106404057-56c61600-646c-11eb-9ae5-1bd344b26da4.png)

`+` is not a `partition function` .